### PR TITLE
Update changelog for the new GC and scheduler

### PR DESCRIPTION
**Pull request by osa1**

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -19,6 +19,16 @@
 
   If you encounter such cases, please adjust the type annotation.
 
+* Garbage collection scheduling algorithm is improved once again, and the
```
Comment (crusso) [inline Changelog.md:22]:
```suggestion
* The garbage collection scheduling algorithm is improved once again, and the
```

File: Changelog.md [hunk 2]
```diff
@@ -19,6 +19,16 @@
...
+  default garbage collection algorithm is now mark-compact. This change
+  significantly reduces garbage collection overhead in terms of cycles, and
+  canisters can now have close to 4 GiB live data (previously 2 GiB).
+
+  Previously mark-compact algorithm was behind the flag `--compacting-gc`. This
```
Comment (crusso) [inline Changelog.md:27]:
```suggestion
  Previously, the mark-compact algorithm was behind the optional flag `--compacting-gc`. This
```

File: Changelog.md [hunk 3]
```diff
@@ -19,6 +19,16 @@
 
   If you encounter such cases, please adjust the type annotation.
 
+* Garbage collection scheduling algorithm is improved once again, and the
+  default garbage collection algorithm is now mark-compact. This change
+  significantly reduces garbage collection overhead in terms of cycles, and
```
Comment (crusso) [inline Changelog.md:24]: also in terms of number of gollections, right?

**Review by crusso — APPROVED** [review]
But see suggested changes

File: Changelog.md (hunk 3 above)
Comment (osa1) [reply Changelog.md:24]: Yes, that's what I meant here. In terms of cycles it's actually worse (~30% more cycles), but we do less GC. I'll edit this part.

**Review by rossberg — APPROVED** [review]
LGTM, modulo Claudio's comment.

**Comment by osa1** [comment]
@rossberg @crusso addressed comments, could you take a look again please?

**Review by rossberg — APPROVED** [review]

**Review by crusso — APPROVED** [review]

**Pull request merged**